test/test_device_histogram.cu

```diff
@@ -426,77 +430,107 @@ struct SearchTransform
 template <>
 struct ScaleTransform<float>
 {
-    int   num_levels;  // Number of levels in array
-    float max;         // Max sample level (exclusive)
-    float min;         // Min sample level (inclusive)
-    float scale;       // Bin scaling factor
-
-    void Init(
-        int   _num_levels,  // Number of levels in array
-        float _max,         // Max sample level (exclusive)
-        float _min,         // Min sample level (inclusive)
-        float _scale)       // Bin scaling factor
+  int num_levels; // Number of levels in array
+  float max;      // Max sample level (exclusive)
+  float min;      // Min sample level (inclusive)
+  float scale;    // Bin scaling factor
+
+  void Init(int _num_levels, // Number of levels in array
+            float _max,      // Max sample level (exclusive)
+            float _min)      // Min sample level (inclusive)
+  {
+    this->num_levels = _num_levels;
+    this->max        = _max;
+    this->min        = _min;
+    this->scale      = 1.0f / static_cast<float>(((max - min) / (num_levels - 1)));
+  }
+
+  // Functor for converting samples to bin-ids  (num_levels is returned if sample is out of range)
+  template <typename SampleT>
+  int operator()(SampleT sample)
+  {
+    if ((sample < min) || (sample >= max))
     {
-        this->num_levels = _num_levels;
-        this->max = _max;
-        this->min = _min;
-        this->scale = 1.0f / _scale;
+      // Sample out of range
+      return num_levels;
     }
 
-    // Functor for converting samples to bin-ids  (num_levels is returned if sample is out of range)
-    template <typename SampleT>
-    int operator()(SampleT sample)
-    {
-        if ((sample < min) || (sample >= max))
-        {
-            // Sample out of range
-            return num_levels;
-        }
+    return (int)((((float)sample) - min) * scale);
+  }
+};
+
+#if TEST_HALF_T
+// Scales samples to evenly-spaced bins
+template <>
+struct ScaleTransform<half_t>
+{
+  int num_levels; // Number of levels in array
+  half_t max;     // Max sample level (exclusive)
+  half_t min;     // Min sample level (inclusive)
+  half_t scale;   // Bin scaling factor
+
+  void Init(int _num_levels, // Number of levels in array
+            float _max,      // Max sample level (exclusive)
+            float _min)      // Min sample level (inclusive)
+  {
+    this->num_levels = _num_levels;
+    this->max        = _max;
+    this->min        = _min;
+    this->scale      = static_cast<half_t>(((_max - _min) / static_cast<half_t>(_num_levels - 1)));
+  }
 
-        return (int) ((((float) sample) - min) * scale);
+  // Functor for converting samples to bin-ids  (num_levels is returned if sample is out of range)
+  template <typename SampleT>
+  int operator()(SampleT sample)
+  {
+    if ((sample < min) || (sample >= max))
+    {
+      // Sample out of range
+      return num_levels;
     }
-};
 
+    return static_cast<int>((static_cast<half_t>(sample) - min) / scale);
```

Comment:
Could you elaborate on why we do not apply `float` logic here?

Reply:
Sorry, I just had to forget to remove the old "dead" code path here. Removed it now.